src/Utilities/objective_container.jl

```diff
@@ -118,8 +118,20 @@ end
 
 function MOI.get(
     o::ObjectiveContainer{T},
-    ::MOI.ObjectiveFunction{F},
-) where {T,F}
+    attr::MOI.ObjectiveFunction{F},
+) where {
+    T,
+    F<:Union{
+        MOI.VariableIndex,
+        MOI.ScalarAffineFunction{T},
+        MOI.ScalarQuadraticFunction{T},
+        MOI.ScalarNonlinearFunction,
+        MOI.VectorOfVariables,
+        MOI.VectorAffineFunction{T},
+        MOI.VectorQuadraticFunction{T},
+        MOI.VectorNonlinearFunction,
+    },
+}
```

Comment:
Hmm. This breaks JuMP:
https://github.com/jump-dev/JuMP.jl/blob/1ee5faee4335386435e3211b8758fba13799234e/test/test_objective.jl#L168-L180

I'm not sure whether we intended this:
```julia
julia> model = GenericModel{Float32}();

julia> @variable(model, x)
x

julia> @objective(model, Min, 2 * x + 1)
2 x + 1

julia> f = objective_function(model, GenericAffExpr{Float64,VariableRef})
2 x + 1

julia> typeof(f)
GenericAffExpr{Float64, GenericVariableRef{Float32}}

julia> g = objective_function(model)
2 x + 1

julia> typeof(g)
GenericAffExpr{Float32, GenericVariableRef{Float32}}
```

Reply:
https://github.com/jump-dev/MathOptInterface.jl/actions/runs/6244194786

Reply:
No I think we meant to use `T` instead of `Float64` in the test